### ISSUE TITLE
implementation of get_oracle_summary function

### ISSUE THREAD

**Issue by pcshih** [weirme/FCSN#15]
https://github.com/weirme/Video_Summary_using_FCSN/blob/0895cccbb2a488369b1bfc7d2c087b3050250898/make_dataset.py#L70

What is the meaning of this function?

**Comment by weirme** [comment]
This function generates a summary from summary of 20 users in the dataset.

**Comment by pcshih** [comment]
The implementation is based on which paragraph of FSCN paper or other paper?

**Comment by weirme** [comment]
Chapter 3.1 of this paper: [Diverse sequential subset selection for supervised video summarization](http://www.cs.utexas.edu/~grauman/papers/nips14_seqdpp.pdf). In my implementation, the greedy algorithm selects the frame marked by the most users each time.

**Comment by pcshih** [comment]
After reading Chapter 3.1, I still cannot realize the process.
Given 3 human summaries with 5 frames:
A: [1,0,1,1,0]
B: [0,0,1,0,0]
C: [0,0,0,1,0]

How to get the final summary?
First: calculate the select times of each frame -> [1,0,2,2,0]
Second: I have no idea...

**Comment by weirme** [comment]
In my implementation, initialize oracle summary as [0, 0, 0, 0, 0], and then pick the most selected frame (here the third), now the oracle summary will be [0, 0, 1, 0, 0]. Determine if the F-score between oracle summary and user summary increases after adding this frame. If true, continue to select next frame,  otherwise it ends.
But it is just my implementation, I didn't find a specific description of the greedy algorithm used in the paper. So I'm not sure if the algorithm is like this.

**Comment by pcshih** [comment]
Where is FCSN mentioned that they use "Diverse sequential subset selection for supervised video summarization" for generating a summary from summary of users?

**Comment by weirme** [comment]
This method is mentioned in supplementary materials of paper [Video Summarization with Long Short-term Memory](https://arxiv.org/pdf/1605.08110.pdf).

**Comment by pcshih** [comment]
After I read the paragraph, I implement it.

https://github.com/pcshih/pytorch-FCSN/blob/7d4f874f6c71d5b279b6e26a6ee4882460230fc9/make_dataset.py#L84

Is my understanding identical to yours?

But the performance is quite bad...

**Comment by weirme** [comment]
Have you print the final F-score between generated oracle summary and user summary?

**Comment by pcshih** [comment]
Did you mean the parameter "best_fscore"?

**Comment by pcshih** [comment]
![best_fscore_1](https://user-images.githubusercontent.com/48461528/62602318-1395f000-b926-11e9-9144-82d223f208fa.PNG)
![best_fscore_2](https://user-images.githubusercontent.com/48461528/62602338-1c86c180-b926-11e9-91fb-ed2bf19265c1.PNG)

It seems slightly different.

**Comment by pcshih** [comment]
https://github.com/KaiyangZhou/pytorch-vsumm-reinforce/blob/fdd03be93f090278424af789c120531e49aefa40/main.py#L164

I found that tvsum use avg but summe use max when evaluating.
After I change summe to max, my result gets better.

But I do not know why to use this method...

![FCSN_1D_summe_eval_max](https://user-images.githubusercontent.com/48461528/62604381-8b661980-b92a-11e9-8371-bce9b2a080bd.PNG)

**Comment by pcshih** [comment]
Could you share the tvsum video on your google drive?
[tvsum](https://webscope.sandbox.yahoo.com/catalog.php?datatype=i&did=72) needs authorization....

**Comment by weirme** [comment]
> https://github.com/KaiyangZhou/pytorch-vsumm-reinforce/blob/fdd03be93f090278424af789c120531e49aefa40/main.py#L164
> 
> I found that tvsum use avg but summe use max when evaluating.
> After I change summe to max, my result gets better.
> 
> But I do not know why to use this method...
> 
> ![FCSN_1D_summe_eval_max](https://user-images.githubusercontent.com/48461528/62604381-8b661980-b92a-11e9-8371-bce9b2a080bd.PNG)

Is this result on SumMe? It seems close to that in paper!

**Comment by weirme** [comment]
> Could you share the tvsum video on your google drive?
> [tvsum](https://webscope.sandbox.yahoo.com/catalog.php?datatype=i&did=72) needs authorization....

Wait a moment, I'm now uploading it...

**Comment by pcshih** [comment]
> > https://github.com/KaiyangZhou/pytorch-vsumm-reinforce/blob/fdd03be93f090278424af789c120531e49aefa40/main.py#L164
> > I found that tvsum use avg but summe use max when evaluating.
> > After I change summe to max, my result gets better.
> > But I do not know why to use this method...
> > ![FCSN_1D_summe_eval_max](https://user-images.githubusercontent.com/48461528/62604381-8b661980-b92a-11e9-8371-bce9b2a080bd.PNG)
> 
> Is this result on SumMe? It seems close to that in paper!

Yes, it is summe.

**Comment by pcshih** [comment]
> > Could you share the tvsum video on your google drive?
> > [tvsum](https://webscope.sandbox.yahoo.com/catalog.php?datatype=i&did=72) needs authorization....
> 
> Wait a moment, I'm now uploading it...

Thank you

**Comment by weirme** [comment]
Here is the [link](https://drive.google.com/open?id=14tcW0T6ps9FQmbqmtHW3OG5lqodCeg78).

**Comment by pcshih** [comment]
Got it. Thank you very much.
Did you figure out ?
https://github.com/KaiyangZhou/pytorch-vsumm-reinforce/blob/fdd03be93f090278424af789c120531e49aefa40/main.py#L164

**Comment by weirme** [comment]
May be it is a default setting in evaluation? I also think it's strange...
And I noticed that selected key frames of videos in summe differ greatly from each user, F-score between generated oracle summary and user summary is only nearly 50%, but that is nearly 70% in tvsum. In this case, getting a summary close to every user seems to be difficult. Is this probably a reason to select max?

**Comment by pcshih** [comment]
I agree with your opinion. Let's take this evaluation method for granted.
I also implement [this paper](https://github.com/pcshih/pytorch-VSLUD) which architecture is based on FCSN but there are some problems...

**Comment by weirme** [comment]
I have not read this paper yet, its architecture looks complicated.

**Comment by pcshih** [comment]
Do you have any idea of FCSN in unsupervised version?

**Comment by weirme** [comment]
No... I skip that part when reading the paper...

**Comment by pcshih** [comment]
Shall we implement that part?

**Comment by weirme** [comment]
I will try to implement it after reading that part, but there may be some problems because my computer at home doesn't have a nvidia gpu :sweat_smile::sweat_smile:

**Comment by pcshih** [comment]
I am counting on you.

**Comment by Pager07** [comment]
> Here is the [link](https://drive.google.com/open?id=14tcW0T6ps9FQmbqmtHW3OG5lqodCeg78).

Thanks for this.
